Framework/Types/inc/MantidTypes/Core/DateAndTime.h

```diff
@@ -147,10 +147,66 @@ class MANTID_TYPES_DLL DateAndTime {
 private:
   /// A signed 64-bit int of the # of nanoseconds since Jan 1, 1990.
   int64_t _nanoseconds;
+
+  /// Max allowed nanoseconds in the time; 2^62-1
+  static constexpr int64_t MAX_NANOSECONDS = 4611686018427387903LL;
+
+  /// Min allowed nanoseconds in the time; -2^62+1
+  static constexpr int64_t MIN_NANOSECONDS = -4611686018427387903LL;
 };
 #pragma pack(pop)
+
+/** Default, empty constructor */
+inline DateAndTime::DateAndTime() : _nanoseconds(0) {}
+
+/** Construct a date from nanoseconds.
+ * @param total_nanoseconds :: nanoseconds since Jan 1, 1990 (our epoch).
+ */
+inline DateAndTime::DateAndTime(const int64_t total_nanoseconds) {
+  // Make sure that you cannot construct a date that is beyond the limits...
+  if (total_nanoseconds > MAX_NANOSECONDS)
+    _nanoseconds = MAX_NANOSECONDS;
+  else if (total_nanoseconds < MIN_NANOSECONDS)
+    _nanoseconds = MIN_NANOSECONDS;
+  else
+    _nanoseconds = total_nanoseconds;
+}
```

Comment:
Reverted my change since it caused linker errors on OSX/`clang`, which I did not want to track down given that this was simply code moved from source to header.